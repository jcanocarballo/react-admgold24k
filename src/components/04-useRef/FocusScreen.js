import React, { useRef } from 'react'
import './focusScreen.css'

export const FocusScreen = () => {

  const inputRef = useRef()

  const handleClick = (e) => {
    inputRef.current.select();
  }

  return (
    <div>
      <h1>Focus Screen</h1>
      <hr></hr>
      <input
        ref= { inputRef }
        className="form-control"
        placeholder="Su nombre"  
      >
      </input>
      <button 
        onClick={ handleClick }
        type="button" className="btn btn-primary mt-3">Aceptar</button>
    </div>
  )
}
