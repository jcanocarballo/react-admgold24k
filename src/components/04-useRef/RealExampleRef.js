import React, { useState } from 'react'
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';
import './focusScreen.css'
export const RealExampleRef = () => {

  const [show, setShow] = useState(false);

  return (
    <div>
      <h1>Real Example Ref</h1>
      <br></br>

      { show && <MultipleCustomHooks></MultipleCustomHooks>}

      <button 
        onClick={
          () => {
            setShow(!show)
          }
        }
        type="button" className="btn btn-primary mt-3">
        Show/Hide
      </button>
    </div>
  )
}
