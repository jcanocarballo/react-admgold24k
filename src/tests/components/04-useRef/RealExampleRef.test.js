import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('Prueba archivo <RealExampleRef />', () => {
  const wrapper = shallow( <RealExampleRef></RealExampleRef> );
  test('debe de mostrarse correctamente', () => { 
    expect( wrapper ).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
  });

  test('debe de mostrarse <MultipleCustomHooks></MultipleCustomHooks>', () => { 
    wrapper.find('button').simulate('click');
    expect( wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  });
});