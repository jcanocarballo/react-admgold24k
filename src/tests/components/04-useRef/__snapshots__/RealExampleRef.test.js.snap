// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba archivo <RealExampleRef /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Real Example Ref
  </h1>
  <br />
  <button
    className="btn btn-primary mt-3"
    onClick={[Function]}
    type="button"
  >
    Show/Hide
  </button>
</div>
`;
